Data/models/PlaylistFolder.swift

```diff
@@ -0,0 +1,122 @@
+
+import Foundation
+import CoreData
+import Shared
+
+private let log = Logger.browserLogger
+
+@objc(PlaylistFolder)
+final public class PlaylistFolder: NSManagedObject, CRUD {
+    @NSManaged public var uuid: String?
+    @NSManaged public var title: String?
+    @NSManaged public var order: Int32
+    @NSManaged public var dateAdded: Date?
+    @NSManaged public var playlistItems: Set<PlaylistItem>?
+    public static let savedFolderUUID = "7B6CC019-8946-4182-ACE8-42FE7B704C43"
+    
+    public class func frc(savedFolder: Bool) -> NSFetchedResultsController<PlaylistFolder> {
+        let context = DataController.viewContext
+        let fetchRequest = NSFetchRequest<PlaylistFolder>()
+        fetchRequest.entity = PlaylistFolder.entity(context)
+        fetchRequest.fetchBatchSize = 20
+        
+        let orderSort = NSSortDescriptor(key: "order", ascending: true)
+        let createdSort = NSSortDescriptor(key: "dateAdded", ascending: false)
+        fetchRequest.sortDescriptors = [orderSort, createdSort]
+        
+        if savedFolder {
+            fetchRequest.predicate = NSPredicate(format: "uuid == %@", savedFolderUUID)
+        } else {
+            fetchRequest.predicate = NSPredicate(format: "uuid != %@", savedFolderUUID)
+        }
+        
+        return NSFetchedResultsController(fetchRequest: fetchRequest, managedObjectContext: context,
+                                          sectionNameKeyPath: nil, cacheName: nil)
+    }
+    
+    public static func addFolder(title: String, uuid: String? = nil, completion: ((_ uuid: String) -> Void)? = nil) {
+        DataController.perform(context: .new(inMemory: false), save: false) { context in
+            var folderId: String
+            if let uuid = uuid, !uuid.isEmpty {
+                folderId = uuid
+            } else {
+                folderId = UUID().uuidString
+            }
+            
+            let playlistFolder = PlaylistFolder(context: context)
+            playlistFolder.title = title
+            playlistFolder.dateAdded = Date()
+            playlistFolder.order = -9999
+            playlistFolder.uuid = folderId
+            
+            PlaylistFolder.saveContext(context)
+            PlaylistFolder.reorderItems(context: context)
+            PlaylistFolder.saveContext(context)
+            
+            DispatchQueue.main.async {
+                completion?(folderId)
+            }
+        }
+    }
+    
+    public static func getOtherFoldersCount(context: NSManagedObjectContext? = nil) -> Int {
+        PlaylistFolder.count(predicate: NSPredicate(format: "uuid != %@", PlaylistFolder.savedFolderUUID), context: context ?? DataController.viewContext) ?? 0
+    }
+    
+    public static func getFolder(uuid: String, context: NSManagedObjectContext? = nil) -> PlaylistFolder? {
+        PlaylistFolder.first(where: NSPredicate(format: "uuid == %@", uuid), context: context ?? DataController.viewContext)
```

Comment:
Same as my previous comment, add a proxy method if you have to pass existing context to it